metals/src/main/scala/scala/meta/internal/builds/SbtBuildTool.scala

```diff
@@ -382,4 +402,33 @@ object SbtBuildTool {
     val prepend = autoImports.mkString("", "\n", "\n")
     prepend + text
   }
+
+  def checkCorrectJavaVersion(
+      workspace: AbsolutePath,
+      userJavaHome: Option[String],
+  ): Boolean = {
+    val bspConfigFile = workspace.resolve(".bsp").resolve("sbt.json")
+    if (bspConfigFile.isFile) {
+      val matchesSbtJavaHome =
+        for {
+          text <- bspConfigFile.readTextOpt
+          json = ujson.read(text)
+          args <- json("argv").arrOpt
+          firstArg <- args.headOption
+          javaArg <- firstArg.strOpt
```

Comment:
Could you check if the executable name is java in case at some point we use something different? If it's not JAva we should not suggest any updates